tests/unit_tests/resume_training_test.py

```diff
@@ -65,6 +66,62 @@ def test_resume_training(self):
         # ASSERT WE START FROM THE RIGHT EPOCH NUMBER
         self.assertTrue(first_epoch_cb.first_epoch == 2)
 
+    def test_resume_run_id_training(self):
+        ckpt_root_dir = ""
+        experiment_name = "test_resume_training"
+
+        experiment_dir = get_checkpoints_dir_path(ckpt_root_dir=ckpt_root_dir, experiment_name=experiment_name)
+        original_dir_count = len(os.listdir(experiment_dir))
+
+        train_params = {
+            "max_epochs": 2,
+            "lr_updates": [1],
+            "lr_decay_factor": 0.1,
+            "lr_mode": "step",
+            "lr_warmup_epochs": 0,
+            "initial_lr": 0.1,
+            "loss": "cross_entropy",
+            "optimizer": "SGD",
+            "criterion_params": {},
+            "optimizer_params": {"weight_decay": 1e-4, "momentum": 0.9},
+            "train_metrics_list": [Accuracy(), Top5()],
+            "valid_metrics_list": [Accuracy(), Top5()],
+            "metric_to_watch": "Accuracy",
+            "greater_metric_to_watch_is_better": True,
+        }
+
+        # Train for 1 more epoch
+        net = LeNet()
+        trainer = Trainer(ckpt_root_dir=ckpt_root_dir, experiment_name=experiment_name)
+        trainer.train(model=net, training_params=train_params, train_loader=classification_test_dataloader(), valid_loader=classification_test_dataloader())
+
+        # Check directory size
+        self.assertEqual(original_dir_count + 1, len(os.listdir(experiment_dir)), "You should have 1 run folder created only after calling `Trainer.train`.")
+
+        # TRAIN FOR 1 MORE EPOCH AND COMPARE THE NET AT THE BEGINNING OF EPOCH 3 AND THE END OF EPOCH NUMBER 2
+        first_epoch_cb = FirstEpochInfoCollector()
+        train_params["run_id"] = get_latest_run_id(checkpoints_root_dir=ckpt_root_dir, experiment_name=experiment_name)
```

Comment:
Maybe also add a test that resumes from a run id that is not the latest one?
This would get us more coverage.